pkgs/by-name/ri/ringboard/package.nix

```diff
@@ -0,0 +1,91 @@
+  version = "0.12.2-unstable-2025-09-23";
+
+  src = fetchFromGitHub {
+    owner = "SUPERCILEX";
+    repo = "clipboard-history";
+    rev = "228a39dd8a9aece0bb06f68ad44906b297270628";
+    hash = "sha256-qA7wwvWnnZHN9edkmubEo37F+peU0LQGo/Zl8FpywuE=";
+  };
+
+  cargoHash = "sha256-MFfuUu/hpb6Uaqe21bvXNKRyJazAL5m+Vw/vAeeDYEk=";
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  buildInputs = [
+    libxkbcommon
+    libGL
+
+    # WINIT_UNIX_BACKEND=wayland
+    wayland
+
+    # WINIT_UNIX_BACKEND=x11
+    xorg.libXcursor
+    xorg.libXrandr
+    xorg.libXi
+    xorg.libX11
+  ];
+
+  buildPhase = ''
+     runHook preBuild
+
+    local flagsArray=("-j $NIX_BUILD_CORES --target x86_64-unknown-linux-gnu --offline --release");
+    concatTo flagsArray cargoBuildFlags;
+
+    echo "Building package: clipboard-history-server"
+    cargo build $flagsArray --package clipboard-history-server --no-default-features --features systemd
+    echo "Building package: clipboard-history-x11"
+    cargo build $flagsArray --package clipboard-history-x11 --no-default-features
+    echo "Building package: clipboard-history-wayland"
+    cargo build $flagsArray --package clipboard-history-wayland --no-default-features
+    echo "Building package: clipboard-history"
+    cargo build $flagsArray --package clipboard-history
+    echo "Building package: clipboard-history-tui"
+    cargo build $flagsArray --package clipboard-history-tui
+    echo "Building package: clipboard-history-egui"
+    cargo build $flagsArray --package clipboard-history-egui
+
+     runHook postBuild
+  '';
+
+  # check needs nightly, see:
+  # https://github.com/SUPERCILEX/clipboard-history/issues/22#issuecomment-3322330559
+  doCheck = false;
+
+  postInstall = ''
+    wrapProgram  $out/bin/ringboard-egui --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath finalAttrs.buildInputs}"
+    install -m 444 -D egui/ringboard-egui.desktop $out/share/applications/ringboard-egui.desktop
+    install -Dm644 logo.jpeg -t $out/share/icons/hicolor/1024x1024/
+    # Initializing a GUI can be quite slow, so GUI clients make their windows invisible when closed rather than completely quitting.
+    # To reopen the window, a special file can be deleted which wakes the GUI via inotify.
+    # If, instead, a new instance of the GUI is opened, this special file is used to first check for a previously running instance of the GUI and kill it if it exists.
+    # https://alexsaveau.dev/blog/projects/performance/clipboard/ringboard/ringboard#gui-startup-latency-and-long-lived-client-windows
+    sed -i "s|Exec=ringboard-egui|Exec=$(echo /bin/sh -c \"ps -p \`cat /tmp/.ringboard/\$USER.egui-sleep 2\> /dev/null\` \> /dev/null 2\>\\\&1 \\\&\\\& exec rm -f /tmp/.ringboard/\$USER.egui-sleep \\\|\\\| exec $out/bin/ringboard-egui\")|g" $out/share/applications/ringboard-egui.desktop
+    sed -i "s|Icon=ringboard|Icon=$out/share/icons/hicolor/1024x1024/logo.jpeg|g" $out/share/applications/ringboard-egui.desktop
```

Comment:
```suggestion
```

Ref other comment about `logo.jpeg`

Reply:
If I just apply your previous suggestion plus remove this line, the program has no icon in the menu.
Presumably because the icon is installed as ``ringboard.jpg`` but the ``.desktop`` has ``ringboard`` without extension.
So either I install the logo without extension, or I modify the ``.desktop``.
Which do you prefer?

Also: wouldn't the icon be found faster with a hard-coded path, instead of having to resolve it dynamically?